resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/Client.scala

```diff
@@ -53,7 +55,8 @@ private[spark] class Client(
     kubernetesClientProvider: SubmissionKubernetesClientProvider,
     initContainerComponentsProvider: DriverInitContainerComponentsProvider,
     kubernetesCredentialsMounterProvider: DriverPodKubernetesCredentialsMounterProvider,
-    loggingPodStatusWatcher: LoggingPodStatusWatcher)
+    loggingPodStatusWatcher: LoggingPodStatusWatcher,
+    sparkJobResourceController: TPRCrudCallsImpl)
```

Comment:
the naming here between the object and the class doesn't match.  I like the name `SparkJobResourceController` better than `TPRCrudCalls`.  Maybe you can rename the class?